docs/guides/ajax.rst

```diff
@@ -1,31 +1,233 @@
+			'day' => date('l'),
+		]);
+
+		return true;
+	}
+
+To fetch its output, use ``ajax.fetchPath('<url_path>', options)``.
+
+.. code-block:: js
+
+	var ajax = new require('elgg/Ajax')();
+
+	ajax.fetchPath('myplugin_time').done(function (output) {
+		alert(output.rfc2822);
+		alert(output.day);
+	});
+
+Notes for paths:
+
+* The 3 hooks (see Actions above) will have type ``path:<url_path>``. In this case, "path:myplugin_time".
+* If the page handler echoes a regular web page, ``output`` will be a string containing the HTML.
+
+Fetching views
+--------------
+
+Consider this view:
+
+.. code-block:: php
+
+	// in myplugin/views/default/myplugin/get_link.php
+
+	if (empty($vars['entity']) || !$vars['entity'] instanceof ElggObject) {
+		return;
+	}
+
+	$object = $vars['entity'];
+	/* @var ElggObject $object */
+
+	echo elgg_view('output/url', [
+		'text' => $object->getDisplayName(),
+		'href' => $object->getUrl(),
+		'is_trusted' => true,
+	]);
 
-.. code:: js
+Since it's a PHP file, we must register it for Ajax first:
+
+.. code-block:: php
+
+	// in myplugin_init()
+	elgg_register_ajax_view('myplugin/get_link');
+
+
+To fetch the view, use ``ajax.fetchView('<view_name>', options)``:
+
+.. code-block:: js
+
+	var ajax = new require('elgg/Ajax')();
+
+	ajax.fetchView('myplugin/get_link', {
+		data: {
+			guid: 123 // querystring
+		},
+	}).done(function (output) {
+		$('.myplugin-link').html(output);
+	});
+
+Notes for views:
+
+* The 3 hooks (see Actions above) will have type ``view:<view_name>``. In this case, "view:myplugin/get_link".
+* ``output`` will be a string with the rendered view.
+* The request data are injected into ``$vars`` in the view.
+* If the request data contains ``guid``, the system sets ``$vars['entity']`` to the corresponding entity or ``false`` if it can't be loaded.
+
+.. warning::
+
+	In ajax views and forms, note that ``$vars`` can be populated by client input. The data is filtered like
+	``get_input()``, but may not be the type you're expecting or may have unexpected keys.
+
+
+Fetching forms
+--------------
+
+Consider we have a form view. We register it for Ajax:
+
+.. code-block:: php
+
+	// in myplugin_init()
+	elgg_register_ajax_view('forms/myplugin/add');
+
+To fetch this using ``ajax.fetchForm('<action_name>', options)``.
+
+.. code-block:: js
+
+	var ajax = new require('elgg/Ajax')();
+
+	ajax.fetchForm('myplugin/add').done(function (output) {
+		$('.myplugin-form-container').html(output);
+	});
+
+Notes for forms:
+
+* The 3 hooks (see Actions above) will have type ``form:<action_name>``. In this case, "form:myplugin/add".
+* ``output`` will be a string with the rendered view.
+* The request data are injected into ``$vars`` in your form view.
+* If the request data contains ``guid``, the system sets ``$vars['entity']`` to the corresponding entity or ``false`` if it can't be loaded.
+
+.. note::
+
+	Request data are passed into the form view only. If you need to add inject parameters into the
```

Comment:
Only the request data are passed to the requested form view (i.e. as a third parameter accepted by `elgg_view_form()`). If you need to pass attributes or parameters of the form element rendered by the ``input/form`` view (i.e. normally passed as a second parameter to ``elgg_view_form()``), use the server-side hook ``view_vars, input/form``.